### PR TITLE
User persisted auth done

### PR DESCRIPTION
**Pull request by AliELSharawy**
user is logged in even refresh or program closed until he log out

### CODE REVIEW

**Comment by NoirFLamme** [comment]
The token is null and the page won't boot up due to the following error:
image
Please resolve what is causing the error and make sure it works on different machines.

File: frontend/store/auth-context.js [hunk 1]
```diff
@@ -10,16 +11,32 @@ const AuthContext = React.createContext({
...
   const logoutHandler = () => {
     setToken("");
+    AsyncStorage.setItem('token',"");
   };
 
+  const isLogIn = async () => {
+      try {
+        let userToken = AsyncStorage.getItem('token');
+        console.log("before user token:", userToken);
+        setToken(await userToken);
```
Comment (ZyadSamy) [inline frontend/store/auth-context.js:30]:
```suggestion
        let savedToken = await AsyncStorage.getItem('token');
        if (savedToken) setToken(await userToken);
```

Comment (ZyadSamy) [reply frontend/store/auth-context.js:30]: This should solve the problem that Ramy mentioned.

File: frontend/store/auth-context.js [hunk 2]
```diff
@@ -10,16 +11,31 @@ const AuthContext = React.createContext({
...
 
   const logoutHandler = () => {
     setToken("");
+    AsyncStorage.setItem('token',"");
   };
 
+  const isLogIn = async () => {
+      try {
+        let savedToken = await AsyncStorage.getItem('token');
+        if (savedToken) setToken(await userToken);
```
Comment (ZyadSamy) [inline frontend/store/auth-context.js:29]:
```suggestion
        if (savedToken) setToken(savedToken);
```

File: frontend/store/auth-context.js (hunk 1 above)
Comment (NoirFLamme) [reply frontend/store/auth-context.js]: The persistence isn't working.

File: frontend/store/auth-context.js (hunk 2 above)
Comment (NoirFLamme) [reply frontend/store/auth-context.js]: Alright! now it functions correctly!
Thanks for the fix!

**Review by NoirFLamme — APPROVED** [review]

**Comment by NoirFLamme** [comment]
Going to change the returned value from getUser and then everything will be functional!

**Review by ZyadSamy — APPROVED** [review]

**Pull request merged**